src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,385 @@
+/// @dev Extends OpenZeppelin's upgradeable Ownable and Pausable patterns to ensure upgradability and pause functionality.
+contract ECDSAStakeRegistry is
+    IERC1271Upgradeable,
+    OwnableUpgradeable,
+    PausableUpgradeable,
+    ECDSAStakeRegistryStorage
+{
+    using SignatureCheckerUpgradeable for address;
+    using CheckpointsUpgradeable for CheckpointsUpgradeable.History;
+
+    /// @dev Constructor to create ECDSAStakeRegistry.
+    /// @param _delegationManager Address of the DelegationManager contract that this registry interacts with.
+    constructor(
+        IDelegationManager _delegationManager
+    ) ECDSAStakeRegistryStorage(_delegationManager) {
+        // _disableInitializers();
+    }
+
+    /// @notice Initializes the contract with the given parameters.
+    /// @param _serviceManager The address of the service manager.
+    /// @param _thresholdWeightBps The threshold weight in basis points.
+    /// @param _quorum The quorum struct containing the details of the quorum thresholds.
+    function initialize(
+        address _serviceManager,
+        uint256 _thresholdWeightBps,
+        Quorum memory _quorum
+    ) external initializer {
+        __ECDSAStakeRegistry_init(_serviceManager, _thresholdWeightBps, _quorum);
+    }
+
+    /// @notice Registers a new operator using a provided signature
+    /// @param _operatorSignature Contains the operator's signature, salt, and expiry
+    function registerOperatorWithSignature(
+        address _operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory _operatorSignature
+    ) external {
+        _registerOperatorWithSig(_operator, _operatorSignature);
+    }
+
+    /// @notice Deregisters an existing operator
+    function deregisterOperator() external {
+        _deregisterOperator(msg.sender);
+    }
+
+    /**
+     * @notice Updates the StakeRegistry's view of one or more operators' stakes adding a new entry in their history of stake checkpoints, 
+     * @dev Queries stakes from the Eigenlayer core DelegationManager contract
+     * @param _operators A list of operator addresses to update
+     */
+    function updateOperators(address[] memory _operators) external {
+        for (uint256 i; i < _operators.length; i++) {
+            _updateOperatorWeight(_operators[i]);
+        }
+    }
+
+    /// @notice Updates the strategies and their weights for the quourum
+    /// @dev Access controlled to the contract owner
+    /// @param _quorum The new quorum configuration
+    function updateQuorumConfig(Quorum memory _quorum) external onlyOwner {
+        _updateQuorumConfig(_quorum);
+    }
+
+    /// @notice Updates the weight an operator must have to join the operator set
+    /// @dev Access controlled to the contract owner
+    /// @param _newMinimumWeight The new weight an operator must have to join the operator set
+    function updateMinimumWeight(uint256 _newMinimumWeight) external onlyOwner {
+        minimumWeight = _newMinimumWeight;
+    }
```

Comment:
probably want an event for this as well

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/194/commits/842b403713e8222ade39eb2b03c7c68b5890c3de